src/components/TodoApp/TodoApp.tsx

```diff
@@ -0,0 +1,307 @@
+
+    if (editedTodo) {
+      try {
+        await updateTodo(editedTodo?.id, { completed });
+        setTodos((prev) => prev.map((todo) => {
+          if (todo.id === todoId) {
+            return {
+              ...todo,
+              completed,
+            };
+          }
+
+          return todo;
+        }));
+      } catch (error) {
+        setError(errorObject.Update);
+      } finally {
+        setIsUpdating(false);
+      }
+    }
+  };
+
+  const [isToggleAll, setIsToggleAll] = useState(false);
+
+  const onToggleAll = async () => {
+    setIsToggleAll(true);
+    setError(null);
+    try {
+      await Promise.all(
+        todos.map((todo) => {
+          if (!todo.completed) {
+            updateTodo(todo.id, { completed: true });
+          }
+
+          if (completedTodosId.length === todos.length) {
+            updateTodo(todo.id, { completed: !todo.completed });
+          }
+
+          return todo;
+        }),
+      );
+
+      if (errorType === null) {
+        setTodos((prev) => prev.map((todo) => {
+          if (!todo.completed) {
+            return {
+              ...todo,
+              completed: true,
+            };
+          }
+
+          if (completedTodosId.length === todos.length) {
+            return {
+              ...todo,
+              completed: !todo.completed,
+            };
+          }
+
+          return todo;
+        }));
+      }
+    } catch (error) {
+      setError(errorObject.Update);
+    } finally {
+      setTimeout(() => {
+        setIsToggleAll(false);
+      }, 300);
+    }
+  };
+
+  const { id } = currentUser;
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const response = await getTodos(id);
+        const todosFromServer = response;
+
+        setError(null);
+        setTodos(todosFromServer);
+      } catch (error) {
+        setError(errorObject.Add);
+        Promise.reject();
+      }
+    };
+
+    fetchData();
+  }, []);
+
+  const [status, setStatus] = useState<Status>(Status.All);
+  const filterTodos = (filter: Status) => {
+    switch (filter) {
+      case Status.Active:
+        return todos.filter(todo => !todo.completed);
+
+      case Status.Completed:
+        return todos.filter(todo => todo.completed);
+
+      default:
+        return todos;
+    }
+  };
+
+  const contextValue = {
+    onDelete,
+    onAdd,
+    onTitleChange,
+    onToggleStatus,
+  };
+
+  const visibleTodos = filterTodos(status);
+
+  if (!USER_ID) {
+    return <UserWarning />;
+  }
+
+  return (
+    <Context.Provider value={contextValue}>
+      <div className="todoapp">
+        <h1 className="todoapp__title">todos</h1>
+
+        <div className="todoapp__content">
+          <Header
+            newTodoTitle={newTodoTitle}
+            setNewTodoTitle={setNewTodoTitle}
+            isActive={isActive}
+            isUpdating={isUpdating}
+            onToggleAll={onToggleAll}
+          />
+
+          {todos.length !== 0 && (
```

Comment:
```suggestion
          {todos.length && (
```

Reply:
Done

Reply:
it cant be because of this, fix it and check out why bug is appear